notebooks/kaggle/store-sales-time-series-forecasting/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,1515 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data['date'] = pd.to_datetime(train_data['date'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 40))\n",
+    "for i, current_num_feature in enumerate(numerical_features):\n",
+    "    plt.subplot(len(numerical_features), 1, i + 1)\n",
+    "    x = train_data.groupby(pd.Grouper(key='date', freq='W'))[current_num_feature].median()\n",
+    "    trend = x.rolling(window=7, center=True, min_periods=5).mean()\n",
+    "    plt.plot(x)\n",
+    "    trend.plot(x=x, linewidth=3)\n",
+    "    plt.xlabel('date')\n",
+    "    plt.ylabel('median ' + current_num_feature)\n",
+    "    plt.title(current_num_feature)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>These trend plots prove guesses about extreme values of 'sales' and 'transactions' numerical features</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Categorical data distribution"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"Number of unique values:  \\n\")\n",
+    "for current_cat_feature in categorical_features:\n",
+    "    print(current_cat_feature, '-', train_data[current_cat_feature].nunique())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cat_features_hist = ['family', 'city', 'locale_name', 'store_nbr']\n",
+    "cat_features_pie = ['state', 'store_type', 'event_type', 'locale', 'transferred', 'cluster']"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(25, 15))\n",
+    "for i, current_cat_column in enumerate(cat_features_pie):\n",
+    "    plt.subplot(2, 3, i + 1)\n",
+    "    n = train_data[current_cat_column].value_counts()\n",
+    "    plt.pie(n, labels=n.index, radius=1800, autopct='%1.1f%%')\n",
+    "    plt.axis('equal')\n",
+    "    plt.title(current_cat_column, color='red')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(30, 40))\n",
+    "for i, cat_feature in enumerate(cat_features_hist):\n",
+    "    plt.subplot(len(cat_features_hist), 1, i + 1)\n",
+    "    plt.xlabel(cat_feature)\n",
+    "    plt.ylabel('count')\n",
+    "    sns.countplot(x=train_data[cat_feature], alpha=0.7)\n",
```

Comment:
In case your x-axis labels overlap, you can rotate them by using `rotation=45` or `rotation=90` parameter.